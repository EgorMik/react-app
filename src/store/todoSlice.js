import { createSlice} from '@reduxjs/toolkit';

    export const filters = {
        ALL: "ALL",
        COMPLETED: "COMPLETED",
        NOT_COMPLETED: "NOT_COMPLETED"
    }
    const localStorageTodos = JSON.parse(localStorage.getItem('todo'));

    if (!localStorageTodos) {
    const initalTodos = {
        todos: [],
        filterBy:filters.ALL
    };
    localStorage.setItem('todo', JSON.stringify(initalTodos));
    }

    
const todoSlice = createSlice({
    name:'todos',
    initialState:  JSON.parse(localStorage.getItem('todo')) !== null
    ? JSON.parse(localStorage.getItem('todo'))
    : { todos: [], filterBy:filters.ALL },
    reducers:{
        addTodo(state, action){
            state.todos.push(action.payload)
            localStorage.setItem('todo', JSON.stringify(state));
       },
        removeTodo(state, action){
            state.todos = state.todos.filter(todo => todo.id !== action.payload.id);
           
            localStorage.setItem('todo', JSON.stringify(state));
        },
        toggleComplete(state, action){
            const toggledTodo = state.todos.find(todo => todo.id === action.payload.id);
            toggledTodo.completed = !toggledTodo.completed;
            localStorage.setItem('todo', JSON.stringify(state));
        },
        filterBy(state, action){
            state.filterBy = action.payload;
        },
        clearComplited(state, action){
            state.todos = state.todos.filter(todo => todo.completed !== action.payload);
            localStorage.setItem('todo', JSON.stringify(state));
        },
        dragAndDropSave(state, action) {
            state.todos = action.payload;
            localStorage.setItem('todo', JSON.stringify(state));
          },
    }
});
export const {addTodo,removeTodo,toggleComplete,filterBy,clearComplited,dragAndDropSave} = todoSlice.actions;
export default todoSlice.reducer;
