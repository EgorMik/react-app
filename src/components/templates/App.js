import {useState} from 'react';
import { useDispatch } from 'react-redux';
import { addTodo } from '../../store/todoSlice';
import Form from '../molecules/Form/Form';
import Page from '../pages/Page';
import TodoList from '../organisms/TodoList/TodoList';
import React from 'react';
import { v4 as uuidv4 } from 'uuid';

function App() {
  const [text, setText] = useState('');
  const [radio, setRadio] = useState(false);
  const dispatch = useDispatch();
  
  const handleAction = () => {
    const newItem = {
      id: uuidv4(),
      text:text,
      completed:radio,
  };
    if(text.trim().length) {
      dispatch(addTodo(newItem));
      setText('');
    }
  }
  const handleRadio = () => {
    setRadio(!radio);
    }
   
  return (
    <div className="container">
      <Page/>
      <Form
        value={text}
        radio={radio}
        updateText={setText}
        handleRadio={handleRadio}
        handleAction={handleAction}
      />
      <TodoList/>
    </div>
  );
}

export default App;
