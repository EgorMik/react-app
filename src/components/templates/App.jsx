import {useState} from 'react';
import { useDispatch } from 'react-redux';
import { addTodo } from '../../store/todoSlice';
import Form from '../molecules/Form/Form';
import Page from '../pages/Page';
import TodoList from '../organisms/TodoList/TodoList';
import React from 'react';

function App() {
  const [text, setText] = useState('');
  const [radio, setRadio] = useState(false);
  const dispatch = useDispatch();

  const handleAction = () => {
    if(text.trim().length) {
      dispatch(addTodo({text,radio}));
      setText('');
    }
  }
  const handleRadio = () => {
    setRadio(!radio);
    }
   
  return (
    <div className="container">
      <Page/>
      <Form
        value={text}
        radio={radio}
        updateText={setText}
        handleRadio={handleRadio}
        handleAction={handleAction}
      />
      <TodoList/>
    </div>
  );
}

export default App;
