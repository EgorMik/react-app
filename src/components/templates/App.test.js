import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import App from "./App";
import store from "../../store";

describe("App", () => {
  it("should renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })});