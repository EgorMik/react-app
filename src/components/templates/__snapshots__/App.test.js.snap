// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="page"
  >
    <p>
      TODO
    </p>
  </div>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="header"
      >
        <input
          checked={false}
          className="checkbox"
          onChange={[Function]}
          onClick={[Function]}
          type="checkbox"
        />
        <input
          className="input"
          onChange={[Function]}
          placeholder="Add your important task"
          type="text"
          value=""
        />
      </div>
      <button
        className="button"
        onClick={[Function]}
      >
        Add todo
      </button>
    </form>
  </div>
  <div
    className="list"
  >
    <div
      className="list_box"
    >
      <ul
        className="todo-list"
        data-rbd-droppable-context-id="0"
        data-rbd-droppable-id="todos"
      />
    </div>
    <div
      className="footer"
    >
      <div>
        <p>
          0
           items left
        </p>
      </div>
      <div>
        <button
          className="active"
          onClick={[Function]}
        >
          All
        </button>
        <button
          className="unactive"
          onClick={[Function]}
        >
          Completed
        </button>
        <button
          className="unactive"
          onClick={[Function]}
        >
          Active
        </button>
      </div>
      <div>
        <button
          className="unactive"
          onClick={[Function]}
        >
          Clear Compleated
        </button>
      </div>
    </div>
  </div>
</div>
`;
